docs/guides/state-management.md

```diff
@@ -94,6 +94,201 @@ You **MUST** use immutable default values _or_ use dataclasses `field` initializ
+
+```python
+from dataclasses import dataclass
+
+@dataclass
+class Address:
+  street: str
+  city: str
+
+@me.stateclass
+class State:
+  address: Address
+  addresses: list[Address]
+```
+
+### Non-Serializable Types
+
+Most types that are not in the serializable types list above cannot be serialized. This includes (but is not limited to):
+
+- Functions, lambdas, and methods
+- File handles, sockets, and I/O objects
+- Database connections and cursors
+- Thread and lock objects
+- Protocol buffers and custom complex objects
+
+**If you get a serialization error**, refer to the [serializable types](#serializable-types) section above for supported types.
+
+### Workarounds for Unsupported Types
+
+If you need to work with types that aren't directly serializable, here are some strategies:
+
+#### 1. Break down into primitives
+
+Extract the data you need into serializable primitive types:
+
+```python
+# ❌ Bad: storing complex object
+@me.stateclass
+class State:
+  api_response: requests.Response  # Not serializable
+
+# ✅ Good: extract the data you need
+@me.stateclass
+class State:
+  response_text: str
+  status_code: int
+  headers: dict[str, str]
+
+def on_click(e: me.ClickEvent):
+  state = me.state(State)
+  response = requests.get("https://api.example.com")
+  state.response_text = response.text
+  state.status_code = response.status_code
+  state.headers = dict(response.headers)
+```
+
+#### 2. Serialize to JSON or bytes
+
+Convert complex objects to JSON strings or bytes:
+
+```python
+@me.stateclass
+class State:
+  config_json: str  # Store as JSON string
+
+def on_load(e: me.LoadEvent):
+  state = me.state(State)
+  complex_config = load_complex_config()
+  state.config_json = json.dumps(complex_config)
+
+def use_config():
+  state = me.state(State)
+  config = json.loads(state.config_json)
+  # Use config...
+```
+
+#### 3. Use Pydantic models
+
+Pydantic models are serializable and can handle complex nested structures:
+
+```python
+from pydantic import BaseModel
+
+class UserProfile(BaseModel):
+  name: str
+  email: str
+  settings: dict[str, str]
+  created_at: datetime
+
+@me.stateclass
+class State:
+  profile: UserProfile  # Pydantic models are serializable
+
+def on_login(e: me.ClickEvent):
+  state = me.state(State)
+  # Pydantic can serialize most Python types
+  state.profile = UserProfile(
+    name="Alice",
+    email="alice@example.com",
+    settings={"theme": "dark"},
+    created_at=datetime.now()
+  )
+```
+
+> **Note**: When using Pydantic models, you get less granular state diffing compared to using primitives directly. This is fine if you don't store a large amount of state data.
+
+### Known Issues
+
+Be aware of these known serialization limitations:
+
+- **Dataclass inheritance**: Classes that inherit from other dataclasses may not serialize correctly. See [issue #659](https://github.com/mesop-dev/mesop/issues/659).
```

Comment:
The info here is not exactly correct. The descriptions you mentioned do no match the issue descriptions. Let's just link to the issue. Just use bulleted list and post the issues in order

- https://github.com/mesop-dev/mesop/issues/565
- https://github.com/mesop-dev/mesop/issues/659
- https://github.com/mesop-dev/mesop/issues/814